core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -386,4 +393,21 @@ public Configuration get() {
       return conf;
     }
   }
+
+  private static class SkipSerializableConfiguration implements Serializable {
+
+    private transient volatile Configuration conf;
+
+    SkipSerializableConfiguration(Configuration conf) {
+      this.conf = conf;
+    }
+
+    public Configuration get() {
+      if (conf == null) {
+        throw new IllegalStateException("Configuration was not serialized on purpose but was not set manually either");
```

Comment:
What about logging a warning and then calling `new Configuration()`?

Reply:
`new Configuration()` would create an empty config object in Hive, without reading all the hive-site.xml values (unlike `new HiveConf()`). Using this empty configuration could then lead to file IO failures or unexpected behaviour. Deciding not to serialize the FileIO config must be followed up by setting the missing config explicitly on the deserializer-side. I would rather throw an exception and catch clients who have failed to do so (and fix those instances), than proceed with creating an empty conf which can then lead to issues.